src/controllers/orderController.js

```diff
@@ -0,0 +1,45 @@
+const { Order, OrderItem } = require('../models/Order');
+const { StatusCodes } = require('http-status-codes');
+const stripe = require('stripe')(process.env.STRIPE_SECRET_KEY);
+const { BadRequestError } = require('../errors');
+
+const createOrder = async (req, res) => {
+  const { orderItems, subtotal, tax, total } = req.body;
+
+  // Create an array of OrderItem objects
+  const orderItemObjects = orderItems.map((item) => new OrderItem(item));
+
+  // Perform necessary checks and validations
+  if (!orderItems?.length) {
+    throw new BadRequestError(
+      'Unable to create order because the order items are missing.'
+    );
+  }
+
+  // Create an Order document in the database
+  const order = await Order.create({
+    user: req.user.userId,
+    orderItems: orderItemObjects,
```

Comment:
In which case this can just be `orderItems: orderItems` (or shorthand, just `orderItems`)

And since the OrderSchema, includes the `OrderItemsSchema`, the values will still undergo validation without us needing to create a separate model/table for them.

Reply:
<img width="568" alt="fixed" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/95896159/2b29c9a7-8fe9-45d1-a839-07b2370c96b4">